clusterloader2/pkg/measurement/common/api_availability_measurement.go

```diff
@@ -0,0 +1,207 @@
+	if status == 200 {
+		a.updateAvailabilityMetrics(true, a.clusterLevelMetrics)
+	} else {
+		a.updateAvailabilityMetrics(false, a.clusterLevelMetrics)
+	}
+}
+
+func (a *apiAvailabilityMeasurement) start(config *measurement.Config, SSHToMasterSupported bool, probeDuration int) error {
+	a.hosts = config.ClusterFramework.GetClusterConfig().MasterIPs
+	if len(a.hosts) < 1 {
+		return goerrors.Errorf("API Server measurement can't execute due to no MasterIPs")
+	}
+
+	k8sClient := config.ClusterFramework.GetClientSets().GetClient()
+	provider, err := util.GetStringOrDefault(config.Params, "provider", config.ClusterFramework.GetClusterConfig().Provider)
+	if err != nil {
+		return err
+	}
+
+	a.isRunning = true
+	a.stopCh = make(chan struct{})
+	a.wg.Add(1)
+
+	go func() {
+		defer a.wg.Done()
+		for {
+			select {
+			case <-a.stopCh:
+				return
+			case <-time.After(time.Duration(probeDuration)):
+				a.updateClusterAvailabilityMetrics(k8sClient)
+				if SSHToMasterSupported {
+					a.updateMasterAvailabilityMetrics(k8sClient, config, provider)
+				}
+			}
+		}
+	}()
+	return nil
+}
+
+// Execute starts the api-server healthz probe end point from start action and
+// collects availability metrics in gather.
+func (a *apiAvailabilityMeasurement) Execute(config *measurement.Config) ([]measurement.Summary, error) {
+	SSHToMasterSupported := config.ClusterFramework.GetClusterConfig().SSHToMasterSupported
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		return nil, err
+	}
+
+	probeFrequency, err := util.GetIntOrDefault(config.Params, "frequency", 1)
+	if err != nil {
+		return nil, err
+	}
+
+	switch action {
+	case "start":
+		if a.isRunning {
+			klog.Infof("%s: measurement already running", a)
+			return nil, nil
+		}
+		return nil, a.start(config, SSHToMasterSupported, probeFrequency)
+	case "gather":
+		err := a.stop(SSHToMasterSupported, probeFrequency)
+		if err != nil {
+			return nil, err
+		}
+		return a.summaries, nil
+	default:
+		return nil, fmt.Errorf("unknown action %v", action)
+	}
+}
+
+func (a *apiAvailabilityMeasurement) createClusterAvailabilitySummary(probeFrequency int) error {
+	clusterAvailabilityMetricsOutput := a.clusterLevelMetrics.buildAPIAvailabilityMetricsOutput(probeFrequency, true, "")
+	content, err := util.PrettyPrintJSON(clusterAvailabilityMetricsOutput)
+	if err != nil {
+		return err
+	} else {
+		a.summaries = append(a.summaries, measurement.CreateSummary(apiAvailabilityName, "json", content))
+		return nil
+	}
+}
+
+func (a *apiAvailabilityMeasurement) createMasterAvailabilitySummary(probeFrequency int) error {
+	createMasterAvailabilitySummaryErrors := errors.NewErrorList()
+	for _, host := range a.hosts {
+		hostAvailabilityMetricsOutput := a.hostLevelMetrics[host].buildAPIAvailabilityMetricsOutput(probeFrequency, false, host)
+		content, err := util.PrettyPrintJSON(hostAvailabilityMetricsOutput)
+		if err != nil {
+			createMasterAvailabilitySummaryErrors.Append(err)
+		} else{
```

Comment:
Please fix gofmt:
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/perf-tests/1162/pull-perf-tests-verify-lint/1285548209413820419/build-log.txt

[both here and in the second file]

Reply:
I ran gofmt -s for these 2 files but this error still got thrown. Is that the right command?

Reply:
gofmt -w -s

Reply:
thank you.. ran the command and pushed again.